### PR TITLE
Bug 2030660: Fix bug where imageStream are not get reconciled

### PR DESCRIPTION
**Pull request by nunnatsa**
Fix by start watching imageStreams. Add an E2E test to validate.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Fix bugzilla bug #2030660
```

### CODE REVIEW

File: hack/check_golden_images.sh [hunk 1]
```diff
@@ -5,6 +5,17 @@ set -ex
...
 if [[ $(${KUBECTL_BINARY} get ssp -n ${INSTALLED_NAMESPACE}) ]]; then
+  # Test image streams
+  [[ $(${KUBECTL_BINARY} get imageStream centos8  -n ${IMAGES_NS} --no-headers | wc -l) -eq 1 ]]
+  [[ "$(${KUBECTL_BINARY} get imageStream centos8  -n ${IMAGES_NS} -o json | jq -cM '.spec.tags[0].from')" == '{"kind":"DockerImage","name":"quay.io/kubevirt/centos8-container-disk-images"}' ]]
+
+  #check that HCO reconciles the image stream
+  ./hack/retry.sh 10 3 "${KUBECTL_BINARY} patch imageStream -n ${IMAGES_NS} centos8 --type=json -p '[{\"op\": \"add\", \"path\": \"/metadata/labels/test-label\", \"value\": \"test\"}]'"
+  sleep 10
+  # HCO expect to remove the test-label label from the image stream
+  ./hack/retry.sh 10 3 "[[ -z $(${KUBECTL_BINARY} get imageStream -n ${IMAGES_NS} centos8 -o jsonpath='{.metadata.labels.test-label}') ]]" "${KUBECTL_BINARY} get imageStream -n ${IMAGES_NS} centos8 -o yaml"
```
Comment (erkanerol) [inline hack/check_golden_images.sh:16]: @nunnatsa  Is this line correct?

Comment (nunnatsa) [reply hack/check_golden_images.sh]: yes. What do you mean?

Comment (erkanerol) [reply hack/check_golden_images.sh]: I understand that retry.sh  is using the last part as debug parameter. there is no issue :+1:

**Pull request merged**